### PR TITLE
Add nvim version check berfore installing

### LINKED ISSUE #578: Neovim Setup
## The bug
image

Am using Debian .. i think it's because nvim version at Debian is old, but not sure of the reason.

### PR DESCRIPTION
**Pull request by adamperkowski**
## Type of Change
- [x] Bug fix
- [x] Hotfix

## Description
image

## Issues / other PRs related
- Resolves #578 

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: core/tabs/applications-setup/Developer-tools/neovim-setup.sh [hunk 1]
```diff
@@ -5,6 +5,13 @@
 gitpath="$HOME/.local/share/neovim"
 
 cloneNeovim() {
+    # lazy.nvim requires nvim >= 0.8.0
+    nvim_version=$(nvim --version | head -n 1 | awk '{print $2}')
+    if [ "$(printf "%s\n" "$nvim_version" "0.8.0" | sort -V | head -n 1)" != "0.8.0" ]; then
+        printf "%b\n" "${RED}Neovim version $nvim_version not supported.${RC}"
+        exit 1
+    fi
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/Developer-tools/neovim-setup.sh:13]: move this into a separate function

Comment (adamperkowski) [reply core/tabs/applications-setup/Developer-tools/neovim-setup.sh:13]: sure

Comment (nnyyxxxx) [reply core/tabs/applications-setup/Developer-tools/neovim-setup.sh:13]: call it at the bottom before the cloning process as well

File: core/tabs/applications-setup/Developer-tools/neovim-setup.sh [hunk 2]
```diff
@@ -54,6 +63,7 @@ linkNeovimConfig() {
 checkEnv
 checkEscalationTool
 installNeovim
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/Developer-tools/neovim-setup.sh:65]: put ``instsallneovim`` under ``checkNeovimVer``

Comment (adamperkowski) [reply core/tabs/applications-setup/Developer-tools/neovim-setup.sh:65]: No. In #578's case the version installed from the repo is < 0.8.0.

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**